pkg/language/reader.go

```diff
@@ -428,6 +429,56 @@ func (r *Reader) ExitClassDecl(ctx *ClassDeclContext) {
 			typ.AddAttribute(memberCtx.GetResult())
 		}
 	}
+
+	if path := annotations.ReferencePath(typ); path != "" {
+		if len(r.inputs) > 1 {
+			panic("refernced service with multiple inputs in undefined")
+		}
+
+		if r.service.Versions().Len() > 1 {
+			panic("cannot infer which version to add reference with multiple versions")
+		}
+
+		input := r.inputs[0]
+		currVersion := r.service.Versions()[0]
+		path = strings.TrimPrefix(path, "/")
+		components := strings.Split(path, "/")
+		referencedServiceName := components[0]
+		referencedVersion := components[1]
+		referencedType := components[2]
```

Comment:
I understand this means that for now subresources are not supported right?

For example:
`/api/clusters_mgmt/v1/cloud_providers/{cloud_provider_id}/regions/{region_id}`